engine/table/src/main/java/io/deephaven/engine/util/WindowCheck.java

```diff
@@ -717,17 +719,24 @@ private RowSet recomputeModified() {
                 final Entry taken = priorityQueue.removeTop();
                 Assert.equals(entry, "entry", taken, "taken");
 
+
+
                 // now scan the rest of the entry, which requires reading from the timestamp source;
                 // this would ideally be done as a chunk, reusing the context
-                long newFirst;
-                for (newFirst = entry.firstRowKey + 1; newFirst <= entry.lastRowKey; ++newFirst) {
-                    // TODO: When we port this to Core, we should use an iterator on the column source to take advantage
-                    // of chunks
-                    final long nanos = inWindowColumnSource.timeStampSource.getLong(newFirst);
-                    if (inWindowColumnSource.computeInWindowUnsafe(nanos)) {
-                        // nothing more to do, we've passed out of the window, note the new nanos for this entry
-                        entry.nanos = nanos;
-                        break;
+                long newFirst = entry.firstRowKey + 1;
+                if (newFirst <= entry.lastRowKey) {
+                    try (CloseablePrimitiveIteratorOfLong timestampIterator =
+                            new ChunkedLongColumnIterator(inWindowColumnSource.timeStampSource,
+                                    RowSetFactory.fromRange(entry.firstRowKey + 1, entry.lastRowKey))) {
```

Comment:
Close the row set, too. Use `RowSequenceFactory` instead.

Reply:
If you use `io.deephaven.engine.table.impl.vector.LongVectorColumnWrapper#LongVectorColumnWrapper(io.deephaven.engine.table.ColumnSource<java.lang.Long>, io.deephaven.engine.rowset.RowSet)`, you can use it's `iterator()` heuristic.

Or just:
```
            if (rowSetSize >= CHUNKED_COLUMN_ITERATOR_SIZE_THRESHOLD) {
                return new ChunkedLongColumnIterator(columnSource, rowSet, DEFAULT_CHUNK_SIZE, rowSet.firstRowKey(), rowSetSize);
            } else {
                return new SerialLongColumnIterator(columnSource, rowSet, rowSet.firstRowKey(), rowSetSize);
            }
```